File: src/codes/Users_Manual/transducer_0.rs
# use autd3::prelude::*;
# 
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
# let autd = Controller::builder([AUTD3::new(Vector3::zeros())])
#    .open(autd3::link::Nop::builder()).await?;
let tr = &autd.geometry[0][0];
let idx = tr.idx();
let position = tr.position();
# Ok(())
# }

File: src/codes/Users_Manual/transducer_0.py
~from pyautd3 import Controller, AUTD3
~from pyautd3.link.audit import Audit
~autd: Controller[Audit] = Controller.builder().add_device(AUTD3([0.0, 0.0, 0.0])).open(Audit.builder())
tr = autd.geometry[0][0]
idx = tr.idx
position = tr.position
rotation = tr.rotation
x_dir = tr.x_direction
y_dir = tr.y_direction
z_dir = tr.z_direction